### DISCUSSION TITLE
Why MAX_RUNNABLE_BLOCK_SIZE=16 in ConsumerWorkService

### DISCUSSION THREAD

**Discussion by miaoweiwei** [rabbitmq/rabbitmq-java-client#813] [General]
It is meaningless to set prefetch count>16 without using thread pool

version:5.15.0

**Comment by michaelklishin** [comment]
It is not meaningless according to benchmarks.

`ConsumerWorkService` is meant to guarantee ordered dispatch of delivered protocol methods *on the given channel*. The decision to pick `MAX_RUNNABLE_BLOCK_SIZE` goes back to 2010. If you have
specific suggestions as to what can be improved and can prove the effectiveness with benchmarks,
we would happily accept a PR.

The value can be set to 128 or 256, the effectiveness of higher prefetch values [hits the law of diminishing returns](https://blog.rabbitmq.com/posts/2014/04/finding-bottlenecks-with-rabbitmq-3-3/).

**Comment by miaoweiwei** [comment]
你好，你的邮件我已收到。苗卫伟

**Comment by michaelklishin** [comment]
Setting the value to 256 produces a low single digit % gain in throughput and a comparable reduction in mean consumer latency. I will submit a PR in a bit.
